### PR TITLE
PR4Misu

### PR DESCRIPTION
**Pull request by ChivuDan**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# Default ignored files
+/shelf/
+/.idea
+/CarrefourTestFramework.iml
```
Comment (mihaivoicu7) [inline .gitignore:4]: Mai tre sa adaugi /target aici

File: src/main/java/Runner.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+import core.InitialSetup;
+import io.appium.java_client.android.AndroidDriver;
+import io.appium.java_client.android.AndroidElement;
+import org.openqa.selenium.remote.DesiredCapabilities;
+import tests.Flows;
+
+//demo class for Browserstack
+public class Runner {
+    public static void main(String[] args) throws MalformedURLException, InterruptedException {
+    String sessionName = "Sign-up + obtainAFG + deleteAccount";
```
Comment (mihaivoicu7) [inline src/main/java/Runner.java:13]: bad formatting

File: src/main/java/Runner.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+import java.net.MalformedURLException;
+
+import core.InitialSetup;
+import io.appium.java_client.android.AndroidDriver;
+import io.appium.java_client.android.AndroidElement;
+import org.openqa.selenium.remote.DesiredCapabilities;
+import tests.Flows;
+
+//demo class for Browserstack
+public class Runner {
```
Comment (mihaivoicu7) [inline src/main/java/Runner.java:11]: what's the idea with this class?

File: src/main/java/Runner.java [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    public static void main(String[] args) throws MalformedURLException, InterruptedException {
+    String sessionName = "Sign-up + obtainAFG + deleteAccount";
+    String sessionName2 = "Log-in + Eshop Basket + logOut";
+
+
+       DesiredCapabilities caps = InitialSetup.initiateCapabilities(sessionName);
```
Comment (mihaivoicu7) [inline src/main/java/Runner.java:17]: split into two methods with suggesting names for caps1 and caps2

File: src/main/java/api/android/Android.java [hunk 5]
```diff
@@ -0,0 +1,8 @@
+package api.android;
+
+import io.appium.java_client.android.AndroidDriver;
+
+public class Android {
```
Comment (mihaivoicu7) [inline src/main/java/api/android/Android.java:5]: ????

File: src/main/java/core/ADB.java [hunk 6]
```diff
@@ -0,0 +1,4 @@
+package core;
+
+public class ADB {
```
Comment (mihaivoicu7) [inline src/main/java/core/ADB.java:3]: ???

File: src/main/java/core/InitialSetup.java [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+        // and desired capabilities defined above
+    }
+    public static AndroidDriver<AndroidElement> initiateDriver(Capabilities caps) throws MalformedURLException {
+        AndroidDriver<AndroidElement> driver = new AndroidDriver<AndroidElement>(new URL("http://hub.browserstack.com/wd/hub"), caps);
+        return driver;
+    }
+    public static void allowPolicies(AndroidDriver driver){
+
+        driver.findElementByAndroidUIAutomator("new UiSelector().resourceId(\"com.android.permissioncontroller:id/permission_allow_foreground_only_button\")").click();
+       /* if(driver.findElementByAndroidUIAutomator(new UiSelector().isClickable("com.carrefourpay.ro:id/btn_close")) {
```
Comment (mihaivoicu7) [inline src/main/java/core/InitialSetup.java:43]: comments?

File: src/main/java/core/MyLogger.java [hunk 8]
```diff
@@ -0,0 +1,4 @@
+package core;
+
+public class MyLogger {
```
Comment (mihaivoicu7) [inline src/main/java/core/MyLogger.java:3]: ?

File: src/main/java/core/TestInfo.java [hunk 9]
```diff
@@ -0,0 +1,4 @@
+package core;
+
+public class TestInfo {
```
Comment (mihaivoicu7) [inline src/main/java/core/TestInfo.java:3]: ?

File: src/main/java/core/UiObject.java [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+
+import java.util.NoSuchElementException;
+
+public class UiObject {
+
+    private String locator;
+
+    UiObject(String locator){
+        this.locator=locator;
+        System.out.println(this.locator);
```
Comment (mihaivoicu7) [inline src/main/java/core/UiObject.java:15]: add more details to logging

File: src/main/java/core/UiObject.java [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+
+    }
+    private boolean isXpath(){
+        return !locator.contains("UiSelector");
```
Comment (mihaivoicu7) [inline src/main/java/core/UiObject.java:19]: is this validation real? 
The following string is an xpath according to this: "MMMuuuiiiEUiSelectorEEasdasdas"

File: src/main/java/core/UiObject.java [hunk 12]
```diff
@@ -0,0 +1,33 @@
...
+    }
+
+    public boolean exists(){
```
Comment (mihaivoicu7) [inline src/main/java/core/UiObject.java:22]: maybe rename this method to "isDisplayed"?

File: src/main/java/core/UiSelector.java [hunk 13]
```diff
@@ -0,0 +1,74 @@
+package core;
+
+public class UiSelector {
+
+    private String locator = "new UiSelector()";
+
+    public UiSelector resourceId(String value) {
+        locator += ".resourceId(\"" + value + "\")";
```
Comment (mihaivoicu7) [inline src/main/java/core/UiSelector.java:8]: Use String.format()
`locator = String.format("%s.resourceId(\"%s\")", locator, value)`

File: src/main/java/core/constants/Credentials.java [hunk 14]
```diff
@@ -0,0 +1,4 @@
+package core.constants;
+
+public class Credentials {
```
Comment (mihaivoicu7) [inline src/main/java/core/constants/Credentials.java:3]: ?

File: src/main/java/core/managers/DriverManager.java [hunk 15]
```diff
@@ -0,0 +1,4 @@
+package core.managers;
+
+public class DriverManager {
```
Comment (mihaivoicu7) [inline src/main/java/core/managers/DriverManager.java:3]: ?

File: src/main/java/core/managers/TestManager.java [hunk 16]
```diff
@@ -0,0 +1,4 @@
+package core.managers;
+
+public class TestManager {
```
Comment (mihaivoicu7) [inline src/main/java/core/managers/TestManager.java:3]: ?

File: src/main/java/core/UiSelector.java [hunk 17]
```diff
@@ -0,0 +1,74 @@
+package core;
+
+public class UiSelector {
```
Comment (mihaivoicu7) [inline src/main/java/core/UiSelector.java:3]: This looks like a builder class.
Check the [builder pattern](https://howtodoinjava.com/design-patterns/creational/builder-pattern-in-java/) and intellij [builder plugin](https://plugins.jetbrains.com/plugin/6585-builder-generator)

File: src/main/java/tests/Flows.java [hunk 18]
```diff
@@ -0,0 +1,170 @@
...
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+
+public class Flows {
+    @Test
+    public static void login(AndroidDriver<AndroidElement> driver, DesiredCapabilities caps) {
+        WebDriverWait wait = new WebDriverWait(driver, 5000);
+
+        if ((driver.findElementByAndroidUIAutomator("new UiSelector().resourceId(\"com.carrefourpay.ro:id/tv_title\")")).isDisplayed()) {
```
Comment (mihaivoicu7) [inline src/main/java/tests/Flows.java:33]: Why not use the UiSelector builder class?

File: src/main/java/tests/Flows.java [hunk 19]
```diff
@@ -0,0 +1,170 @@
...
+        driver.findElementByAndroidUIAutomator("new UiSelector().resourceId(\"com.carrefourpay.ro:id/btn_login_register\")").click();
+        driver.findElementByAndroidUIAutomator("new UiSelector().resourceId(\"com.carrefourpay.ro:id/et_user_id\")").sendKeys("0722812150");
+        driver.findElementByAndroidUIAutomator("new UiSelector().resourceId(\"com.carrefourpay.ro:id/btn_continue\")").click();
+        // driver.findElementByAndroidUIAutomator(UiSelector.className("android.widget.TextView")).sendKeys("123987";
+        wait.until(ExpectedConditions.elementToBeClickable((driver.findElementByAndroidUIAutomator("new UiSelector().resourceId(\"com.carrefourpay.ro:id/et_password\")"))));
+        driver.findElementByAndroidUIAutomator("new UiSelector().resourceId(\"com.carrefourpay.ro:id/et_password\")").sendKeys("123987");
+        driver.findElementByAndroidUIAutomator("new UiSelector().resourceId(\"com.carrefourpay.ro:id/btn_login\")").click();
+        if ((driver.findElementByAndroidUIAutomator("new UiSelector().resourceId(\"com.carrefourpay.ro:id/btn_send\")")).isDisplayed()) {
```
Comment (mihaivoicu7) [inline src/main/java/tests/Flows.java:44]: why not use the helper method?

File: src/main/java/tests/Flows.java [hunk 20]
```diff
@@ -0,0 +1,170 @@
...
+                .getAttribute("text").equals(itemAddedName));}
+        catch (Exception e)
+        {
+            jse.executeScript("browserstack_executor: {\"action\": \"setSessionStatus\", \"arguments\": {\"status\":\"failed\", \"reason\": \"Results not found\"}}");
+        }
+           // jse.executeScript("browserstack_executor: {\"action\": \"setSessionStatus\", \"arguments\": {\"status\": \"passed\", \"reason\": \"Results found!\"}}");
+
+    }
+    @Test
+    public static void logOut(AndroidDriver<AndroidElement> driver, DesiredCapabilities caps){
```
Comment (mihaivoicu7) [inline src/main/java/tests/Flows.java:87]: this @test has no assert

File: target/maven-status/maven-compiler-plugin/compile/default-compile/createdFiles.lst [hunk 21]
```diff
@@ -0,0 +1,12 @@
+api\android\Android.class
```
Comment (mihaivoicu7) [inline target/maven-status/maven-compiler-plugin/compile/default-compile/createdFiles.lst:1]: target files should not be commited

**Review by mihaivoicu7 — COMMENTED** [review]
not a lot of code.. looks like everything except Flows.java is extra

**Review by ChivuDan — COMMENTED** [review]
Addressed

**Pull request closed without merging**